client/state/sites/plans/selectors.js

```diff
@@ -201,3 +201,31 @@ export function isCurrentUserCurrentPlanOwner( state, siteId ) {
 
 	return get( currentPlan, 'userIsOwner', false );
 }
+
+/**
+ * Returns a site's current plan's product slug
+ *
+ * @param  {Object}  state   Global State tree
+ * @param  {Number}  siteId  Site ID
+ * @return {String}          The site's current plan's product slug
+ */
+export function getSitePlanSlug( state, siteId ) {
+	return get( getCurrentPlan( state, siteId ), 'productSlug', null );
+}
+
+// Duplicated from lib/plans. Proper solution in https://github.com/Automattic/wp-calypso/pull/9635
+function planHasFeature( plan, feature ) {
+	return includes( get( PLANS_LIST[ plan ], 'getFeatures', () => [] )(), feature );
+}
+
+/**
+ * Whether a site's current plan includes a given feature
+ *
+ * @param  {Object}  state   Global State tree
+ * @param  {Number}  siteId  Site ID
+ * @param  {String}  feature The feature we're looking for
+ * @return {Boolean}         True if the site's current plan includes the feature
+ */
+export function hasFeature( state, siteId, feature ) {
```

Comment:
Personally I would add helper functions that would combine `hasFeature` and for example `FEATURE_UNLIMITED_PREMIUM_THEMES` into something like `hasUnlimitedPremiumThemesFeature` for a nice clean solution.